SCons/Tool/MSCommon/vc.py

```diff
@@ -357,20 +432,23 @@ def _make_target_host_map(all_hosts, host_all_targets_map):
     host_all_hosts = OrderedDict([
         ('amd64', ['amd64', 'x86']),
         ('x86',   ['x86']),
+        ('arm64', ['amd64', 'x86']),
```

Comment:
Can't you build arm64 on an arm 64 host?

Reply:
> Can't you build arm64 on an arm 64 host?

The `host_all_hosts` dictionary is for configuring the "native" architecture of the Visual Studio *tools* rather than the target architectures.  `host_all_targets` and `host_def_targets` are used to defined target architectures.

That particular stanza is for VS2005-VS2015.  On arm64 for VS2005-VS2015, the VS tools will either be amd64 (64-bit) or x86 (32-bit).

Native arm64 VC tools were introduced in VS2022 and only for the toolset introduced in VS2022 (i.e., 14.3X).  For VS2005-VS2015, prefer 64-bit amd64 tools if available and then 32-bit x86 tools.

Reply:
Gotcha. Thanks for explaining.